src/celeritas/neutron/data/NeutronElasticData.hh

```diff
@@ -0,0 +1,189 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/neutron/data/NeutronElasticData.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "corecel/Macros.hh"
+#include "corecel/Types.hh"
+#include "corecel/data/Collection.hh"
+#include "corecel/math/Quantity.hh"
+#include "celeritas/Quantities.hh"
+#include "celeritas/Types.hh"
+#include "celeritas/grid/GenericGridData.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Parameters for sampling the momentum transfer of CHIPS neutron-nucleus
+ * elastic scattering.
+ */
+struct ExchangeParameters
+{
+    using Real4 = Array<real_type, 4>;
+
+    // Momentum-dependent parameters in G4ChipsNeutronElasticXS
+    real_type ss{};  // square slope of the first diffractive maximum
+    Real4 slope{0, 0, 0, 0};  //!< slope of CHIPS diffractive maxima
+    Real4 expnt{0, 0, 0, 0};  //!< mantissa of CHIPS diffractive maxima
+};
+
+//---------------------------------------------------------------------------//
+/*!
+ * A-dependent data for the differential cross section (momentum transfer) of
+ * the CHIPS neutron-nucleus elastic model.
+ */
+struct ChipsDiffXsCoefficients
+{
+    using ChipsArray = Array<real_type, 42>;
+
+    // Coefficients
+    ChipsArray par{};  //!< Coefficients as a function of atomic mass numbers
+};
+
+//---------------------------------------------------------------------------//
+/*!
+ * Model and particles IDs for neutron--nucleus elastic scattering.
+ */
+struct NeutronElasticIds
+{
+    ActionId action;
+    ParticleId neutron;
+
+    //! Whether the IDs are assigned
+    explicit CELER_FUNCTION operator bool() const { return action && neutron; }
+};
+
+//---------------------------------------------------------------------------//
+/*!
+ * Microscopic cross section data for the neutron elastic interaction.
+ */
+template<Ownership W, MemSpace M>
+struct NeutronElasticXsData
+{
+    using XsUnits = units::Native;  // [len^2]
+
+    template<class T>
+    using Items = Collection<T, W, M>;
+    template<class T>
+    using ElementItems = Collection<T, W, M, ElementId>;
+
+    //// MEMBER DATA ////
+
+    Items<real_type> reals;
+    ElementItems<GenericGridData> elements;
```

Comment:
Any reason to have a separate struct for this, or could this go directly in `NeutronElasticData`?